compiler/utils/astrepr.nim

```diff
@@ -16,63 +35,120 @@ import
+    trfShowNodeComments ## `.comment` field
+    trfShowNodeErrors ## Embedded `nkError` reports
+    trfShowNodeTypes ## Flat render of the node type
+
+  TReprConf* = object
+    ## Main debug configuration object
+    flags*: set[TReprFlag]  ## Set of the formatting options for the
+    ## procedure. Several built-in constants are defined in this module
+    ## that might be suitable for different debugging or AST exploration
+    ## tasks.
+    maxDepth*: int ## Ignore all nodes that are placed deeper than that.
+    ## Useful to see a high-level overview for large nodes, such as full
+    ## proc bodies
+    maxLen*: int ## on each level, print upto that number of subnodes.
+    ## Useful to cut out parts on large `case` trees, toplevel nodes for
+    ## modules, statement lists and such.
+    maxPath*: int ## maximum path length for subnodes
 
-    trfShowSymFlags
-    trfShowSymLineInfo
-    trfShowSymTypes
-    trfShowSymMagic
-    trfShowSymKind
-    trfShowSymOwner
-    trfShowSymOptions
-    trfShowSymPosition
-    trfShowSymOffset
-    trfShowSymBitsize
-    trfShowSymAlignment
-
-    trfShowTypeCallConv
-    trfShowTypeFlags
-    trfShowTypeSize
-    trfShowTypeOwner
-    trfShowTypeBaseId
-
-    trfShowNodeLineInfo
-    trfShowNodeFlags
-    trfShowNodeIds
-    trfShowNodeValue
-    trfShowNodeComments
-    trfShowNodeErrors
-    trfShowNodeTypes
 
 
 export colortext.`$`
 
-const treeReprAllFields* = { trfShowSymFlags .. trfShowNodeTypes }
-
-const defaultTreeReprFlags* = {
-  trfPositionIndexed,
-  trfReportInfo,
-  trfSkipAuxError,
-  trfIndexVisisted
-} + treeReprAllFields - {
-  trfShowNodeLineInfo,
-  trfShowSymLineInfo,
-  trfShowSymOptions,
-}
+const treeReprAllFields* = { trfShowSymFlags .. trfShowNodeTypes } ## Set
+                           ## of flags to print all fields in all tree
+                           ## reprs
+
+const style = (
+  kind: fgBlue,
+  nilIt: fgRed,
+  ident: fgCyan,
+  setIt: termFg(2, 1, 3),
+  number: fgCyan,
+  strLit: fgYellow,
+  floatLit: fgMagenta,
+  dim: termFg(15),
+  errKind: termFg(5, 2, 0),
+  err: fgRed,
+  comment: termFg(4, 2, 1),
+  owner: termFg(1, 3, 0)
+)
+
+const defaultTReprConf* = TReprConf(
+  maxDepth: 120,
+  maxLen: 30,
+  maxPath: 1,
+  flags: {
+    trfReportInfo,
+    trfSkipAuxError,
+    trfIndexVisisted
+  } + treeReprAllFields - {
+    trfShowNodeLineInfo,
+    trfShowSymLineInfo,
+    trfShowSymOptions,
+    trfShowTypeBaseId,
+    trfShowTypeAlloc
+  }
+) ## Default based configuration for the tree repr printing functions
+
+const compactTReprConf* =
+  block:
+    var base = defaultTReprConf
+    base.flags.incl trfPackedFields
+    base
+  ## Compacted tree repr configuration
+
+var implicitTReprConf*: TReprConf = defaultTReprConf ## global
+  ## configuration object that is implicitly used by `debugAst` and
+  ## `debugType`. Can be used in order to configure behaviour of the
+  ## debugging functions that could later be called from `gdb` environment
```

Comment:
This is so on the money, the Python gdb and lldb scripts have to do too much, we really just need to compile in the procs and those scripts should be very simple shims.